oracle/types/oracle.go

```diff
@@ -24,116 +23,121 @@ type (
 
 	// PriceWebSocketQueryHandler is a type alias for the price web socket query handler.
 	// This is responsible for building the web socket query for the price provider and
 	// returning the resolved and unresolved prices.
-	PriceWebSocketQueryHandler = wshandlers.WebSocketQueryHandler[mmtypes.Ticker, *big.Int]
+	PriceWebSocketQueryHandler = wshandlers.WebSocketQueryHandler[ProviderTicker, *big.Float]
 
 	// PriceResponse is a type alias for the price response. A price response is
 	// composed of a map of resolved prices and a map of unresolved prices. Resolved
 	// prices are the prices that were successfully fetched from the API, while
 	// unresolved prices are the prices that were not successfully fetched from the API.
-	PriceResponse = providertypes.GetResponse[mmtypes.Ticker, *big.Int]
+	PriceResponse = providertypes.GetResponse[ProviderTicker, *big.Float]
 
 	// ResolvedPrices is a type alias for the resolved prices.
-	ResolvedPrices = map[mmtypes.Ticker]providertypes.ResolvedResult[*big.Int]
+	ResolvedPrices = map[ProviderTicker]providertypes.ResolvedResult[*big.Float]
 
 	// UnResolvedPrices is a type alias for the unresolved prices.
-	UnResolvedPrices = map[mmtypes.Ticker]providertypes.UnresolvedResult
+	UnResolvedPrices = map[ProviderTicker]providertypes.UnresolvedResult
 
 	// TickerPrices is a type alias for the map of prices. This is a map of tickers i.e.
 	// BTC/USD, ETH/USD, etc. to their respective prices.
-	TickerPrices = map[mmtypes.Ticker]*big.Int
+	TickerPrices = map[ProviderTicker]*big.Float
+
+	// AggregatorPrices is a type alias for a map of off-chain ticker to the price. It is
+	// critical to note that the aggregator tracks prices indexed by off-chain tickers. When
+	// a final price is calculated (after aggregation), it is indexed by the canonical on-chain
+	// ticker.
+	AggregatorPrices = map[string]*big.Float
 
 	// PriceAggregator is a type alias for the price aggregator. This is responsible for
-	// aggregating the resolved prices from the price providers.
-	PriceAggregator = aggregator.Aggregator[string, TickerPrices]
+	// aggregating the resolved prices from the price providers. This maps the
+	// provider -> off-chain-ticker -> price.
+	PriceAggregator = aggregator.Aggregator[string, AggregatorPrices]
 
 	// PriceAggregationFn is a type alias for the price aggregation function. This function
 	// is used to aggregate the resolved prices from the price providers.
-	PriceAggregationFn = aggregator.AggregateFn[string, TickerPrices]
+	PriceAggregationFn = aggregator.AggregateFn[string, AggregatorPrices]
 
 	// AggregatedProviderPrices is a type alias for the aggregated provider prices. This is
 	// a map of provider names to their respective ticker prices.
-	AggregatedProviderPrices = aggregator.AggregatedProviderData[string, TickerPrices]
+	AggregatedProviderPrices = aggregator.AggregatedProviderData[string, AggregatorPrices]
```

Comment:
`map[string]map[string]*big.Float` 

Does it make sense to move all of this stuff into `aggregator/aggregator.go` so we don't have to consult multiple files to figure out this is just a nested map?